agent/drive_handler.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+)
+
+type drive struct {
+	Name       string `json:"name"`
+	DriveID    string `json:"-"`
+	MajorMinor string `json:"maj:min"`
+	Size       string `json:"size"`
+	Type       string `json:"type"`
+}
+
+type drives struct {
+	BlockDevices []drive `json:"blockdevices"`
+}
+
+type driveHandler struct {
+	drives drives
+}
+
+// discoverDrives will use lsblk to populate the drives field in
+// the drive handler.
+func (d *driveHandler) discoverDrives() error {
+	ds, err := lsblk()
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(ds.BlockDevices); i++ {
+		if !isStubDrive(ds.BlockDevices[i]) {
+			continue
+		}
+
+		id := getDriveID(ds.BlockDevices[i])
+		ds.BlockDevices[i].DriveID = id
+	}
+
+	fmt.Println(ds)
+	d.drives = ds
+	return nil
+}
+
+// lsblk will call the lsblk linux command and parse the given output into an
```

Comment:
I'm somewhat hesitant for us to add a dependency on lsblk to the Agent specifically as that is adding a requirement to the VM image. I think in general it would be preferable to keep hard requirements for that image as barebones as possible in order to encourage lightweight images (and simplify image creation).

lsblk appears to be part of the util-linux package, which is fairly common to install but isn't necessarily always present by default. [Busybox for example doesn't appear to have lsblk](https://busybox.net/downloads/BusyBox.html).

It seems like we are just using lsblk to get a list of block devices that have a given size (512B), which may be fairly straightfoward to instead do ourselves:
1. List of blocks devices can be obtained by listing the `/sys/class/block` dir
2. Size is just reading `/sys/class/block/<diskname>/size`

Reply:
I went ahead and updated, but rather than using `/sys/class/block` I used `/sys/block`, since we aren't interested in partitions